helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -90,12 +90,16 @@
 
   private final HelixZkClient _zkClient;
   private final ConfigAccessor _configAccessor;
+  // true if ZKHelixAdmin was instantiated with a HelixZkClient, false otherwise
+  // This is used for close() to determine how ZKHelixAdmin should close the underlying ZkClient
+  private boolean _usesExternalZkClient;
 
   private static Logger logger = LoggerFactory.getLogger(ZKHelixAdmin.class);
 
   public ZKHelixAdmin(HelixZkClient zkClient) {
```

Comment:
Please mark this constructor as @Deprecated.

Reply:
Done